lib-api/src/main/java/org/ergoplatform/appkit/BoxSelectorsJavaHelpers.scala

```diff
@@ -26,9 +28,13 @@ object BoxSelectorsJavaHelpers {
       .map(InputBoxWrapper.apply).toIterator
     val targetAssets = tokensToSpend.convertTo[mutable.LinkedHashMap[ModifierId, Long]].toMap
     val foundBoxes: IndexedSeq[InputBox] = DefaultBoxSelector.select(inputBoxes, amountToSpend, targetAssets) match {
-      case Left(err) => 
-        throw new RuntimeException(
-          s"Not enough funds in boxes to pay $amountToSpend nanoERGs, \ntokens: $tokensToSpend, \nreason: $err")
+      case Left(err) =>
+        if (err.isInstanceOf[NotEnoughCoinsForChangeBoxesError]) {
+          throw new InputBoxesSelectionException.NotEnoughCoinsForChangeException(err.message)
```

Comment:
Why repackage original exception in a new one?

Reply:
Idea is that we have different Java exceptions for all three error types that can occur so that clients can determine which error was thrown by the Scala code. The Java Exceptions for NotEnoughErgsException and NotEnoughTokensException should also provide fields like the ones added to the Scala Errors in ergoplatform/ergo#1423 - I did not add them here though, because it has nothing to do with the problem this PR tries to solve.